src/BSON/Timestamp.c

```diff
@@ -226,6 +232,102 @@ PHP_METHOD(Timestamp, __wakeup)
 }
 /* }}} */
 
+/* {{{ proto string Timestamp::serialize()
+*/
+PHP_METHOD(Timestamp, serialize)
+{
+	php_phongo_timestamp_t   *intern;
+#if PHP_VERSION_ID >= 70000
+	zval                      retval;
+#else
+	zval                     *retval;
+#endif
+	php_serialize_data_t      var_hash;
+	smart_str                 buf = { 0 };
+	char                      s_increment[24];
+	char                      s_timestamp[24];
+	int                       s_increment_len;
+	int                       s_timestamp_len;
+
+	intern = Z_TIMESTAMP_OBJ_P(getThis());
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	s_increment_len = snprintf(s_increment, sizeof(s_increment), "%" PRId64, intern->increment);
+	s_timestamp_len = snprintf(s_timestamp, sizeof(s_timestamp), "%" PRId64, intern->timestamp);
```

Comment:
I just realized that `php_phongo_timestamp_init_from_string()` is using `int64_t` for these, which is incorrect. I must have missed that when reviewing https://github.com/mongodb/mongo-php-driver/pull/386/files#diff-3598851e9e7e7e1da5e874e6bbfdb449R72.